llvm/lib/TargetParser/RISCVTargetParser.cpp

```diff
@@ -58,6 +58,19 @@ bool hasFastVectorUnalignedAccess(StringRef CPU) {
   return Info && Info->FastVectorUnalignedAccess;
 }
 
+bool hasValidCPUModel(StringRef CPU) {
+  const CPUModel CPUModel = getCPUModel(CPU);
+  return CPUModel.MVendorID != 0 && CPUModel.MArchID != 0 &&
```

Comment:
Can you add a test for this error maybe? I think maybe the generic cpus will never have IDs so should be fine for tests.

Reply:
Done!